opennlp-tools/src/main/java/opennlp/tools/lemmatizer/DictionaryLemmatizer.java

```diff
@@ -47,26 +47,34 @@
    * @param dictionary
    *          the input dictionary via inputstream
    */
+  // To have duplicate keys we need to have a collection for values too,
+  // this way:
+  // 1. We could get every lemma for a word,pos pair in the key
+  // 2. We could get every pos,lemma for a word in the key
+  // Crucially, both keys and values need to be collections, probably lists
   public DictionaryLemmatizer(final InputStream dictionary) {
     this.dictMap = new HashMap<>();
-    final BufferedReader breader = new BufferedReader(new InputStreamReader(dictionary));
+    final BufferedReader breader = new BufferedReader(
+        new InputStreamReader(dictionary));
     String line;
     try {
       while ((line = breader.readLine()) != null) {
         final String[] elems = line.split("\t");
-        this.dictMap.put(Arrays.asList(elems[0], elems[1]), elems[2]);
+        this.dictMap.put(Arrays.asList(elems[0], elems[1]),
+            Arrays.asList(elems[2]));
```

Comment:
use Collections.singletonList(elems[2]) instead if there's only gonna be a single element always